qdrant_test/collections_test.go

```diff
@@ -0,0 +1,188 @@
+package qdrant_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/qdrant/go-client/qdrant"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCollectionsClient(t *testing.T) {
+	collectionName := t.Name()
+	vectorSize := uint64(384)
+	distance := qdrant.Distance_Cosine
+	apiKey := "<COLLECTIONS_TEST>"
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	container, err := distributedQdrant(ctx, apiKey)
+	require.NoError(t, err)
+
+	err = container.Start(ctx)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		err := container.Terminate(ctx)
+		require.NoError(t, err)
+	})
```

Comment:
Can it be that the context is cancelled here already (see line 18), or is the order of the calls to cleanup garantied?

Reply:
I checked the documentation, it's fine
```
the order of calls to t.Cleanup is guaranteed, but it's important to note that they are executed in last-in, first-out (LIFO) order. This means that the cleanup functions are executed in the reverse order of how they were registered.
```

Reply:
Yes. I switched this to use `defer` too. Worked fine.
I think we can continue with `t.CleanUp` since we get to test for the failure for `container.Terminate` as well.